### PR TITLE
Ideaspace/strapiv4

### PR DESCRIPTION
**Pull request by dbradham**
- This includes some basic changes to access strapi v4 backend as well as the implementation of ideas in the shared agent

### CODE REVIEW

File: packages/utility/agent.ts [hunk 1]
```diff
@@ -118,7 +118,10 @@ const Opportunities = {
   list: () => requests.get<Opportunity[]>("opportunities"),
 };
 
-const Ideas = {};
+const Ideas = {
+  get: async (params?: URLSearchParams) => 
+    requests.get<Idea[]>("idea-cards", params).then(res => res)
```
Comment (Enjoy2Live) [inline packages/utility/agent.ts:123]: .then here is redundant

Comment (dbradham) [reply packages/utility/agent.ts]: Thanks. Just updated to remove this and support more pages within IdeaSpace on v4

**Comment by alicefate** [comment]
I think DashboardPage.js need to change axios, DashboardPage/Ideas/Ideas.js need to change published_at and updated_at

File: packages/utility/agent.ts [hunk 2]
```diff
@@ -118,7 +118,14 @@ const Opportunities = {
...
 };
 
-const Ideas = {};
+const Ideas = {
+  get: async (params?: URLSearchParams) => 
+    requests.get<Idea[]>("idea-cards", params),
+  getIdea: async (id: string, params?: URLSearchParams) => 
+    requests.get<Idea>(`/idea-cards/${id}`, params),
+  post: async (url: string, body: {}) =>
+    requests.post<Idea>(url, body)
```
Comment (Enjoy2Live) [inline packages/utility/agent.ts:127]: I don't think having these as async functions make sense, they're going to be awaited in the app level.

File: packages/models/index.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
-export type { Skill } from './skill';
-export type { User } from './user';
+export { Expectation } from "./expectation";
+export { SkillLevel } from "./level";
+export { NewApplicant } from "./newApplicant";
+export { Opportunity } from "./opportunity";
+export { HeroImage, Project } from "./project";
+export { Skill } from "./skill";
+export { User } from "./user";
+export { Idea } from './idea';
```
Comment (Enjoy2Live) [inline packages/models/index.ts:8]: why did we change this from `export type`  to just `export` ?

Comment (dbradham) [reply packages/models/index.ts]: Thanks. Not sure this was intentional.

File: packages/utility/agent.ts (hunk 2 above)
Comment (dbradham) [reply packages/utility/agent.ts]: Okay. I have removed the async identifier here and tested that things are still working.

**Comment by dbradham** [comment]
> I think DashboardPage.js need to change axios, DashboardPage/Ideas/Ideas.js need to change published_at and updated_at

Hey @alicefate could you leave a comment on the files & lines where you would propose this change?

File: packages/models/idea.ts
[the remark this answers is not in the document]
Comment (dbradham) [reply packages/models/idea.ts:1]: We need to add these attributes to strapiv4, and then we can add them to the front end codebase. Right now they only exist in the legacy strapi.

**Comment by alicefate** [comment]
> > I think DashboardPage.js need to change axios, DashboardPage/Ideas/Ideas.js need to change published_at and updated_at
> 
> Hey @alicefate could you leave a comment on the files & lines where you would propose this change?

@dbradham The first you already changes. The second is line 22 and line 27 in dev-launchers-platform/apps/ideaspace/src/components/modules/DashboardPage/Ideas/Ideas.js /

**Comment by dbradham** [comment]
> > > I think DashboardPage.js need to change axios, DashboardPage/Ideas/Ideas.js need to change published_at and updated_at
> > 
> > 
> > Hey @alicefate could you leave a comment on the files & lines where you would propose this change?
> 
> @dbradham The first you already changed. The second is line 22 and line 27 in dev-launchers-platform/apps/ideaspace/src/components/modules/DashboardPage/Ideas/Ideas.js /

Thanks!

How exactly should these lines be updated? I'm not seeing anything obviously wrong from my end, but maybe I am missing something?
image

**Comment by alicefate** [comment]
> > > > I think DashboardPage.js need to change axios, DashboardPage/Ideas/Ideas.js need to change published_at and updated_at
> > > 
> > > 
> > > Hey @alicefate could you leave a comment on the files & lines where you would propose this change?
> > 
> > 
> > @dbradham The first you already changed. The second is line 22 and line 27 in dev-launchers-platform/apps/ideaspace/src/components/modules/DashboardPage/Ideas/Ideas.js /
> 
> Thanks!
> 
> How exactly should these lines be updated? I'm not seeing anything obviously wrong from my end, but maybe I am missing something? image

line 22 should be publishedAt, and line 27 should be updatedAt. They are directly from database, and I saw you change it in other files.

**Review by pyxld-kris — APPROVED** [review]

**Review by pyxld-kris — DISMISSED** [review]

**Pull request merged**